### ISSUE TITLE
Can't compile on GNU toolchain for windows

### ISSUE THREAD

**Issue by Cy-Tek** [rust-onig/rust-onig#130]
Hello,

I'm having an error when trying to compile this library on the gnu toolchain for windows. It works fine for msvc, but I get the following error otherwise.

```
error: linking with `gcc` failed: exit code: 1
  |
  = note: "gcc" "-Wl,--enable-long-section-names" "-fno-use-linker-plugin" "-Wl,--nxcompat" "-nostdlib" "-m64" "C:\\Users\\JoshHannaford\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu:\\Users\\JoshHannaford\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deJoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\meteormd-591af4f6383d7032.1z96j2ryhf3f63q4.rcgu.o" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\metnaford\\CLionProjects\\meteormd\\target\\debug\\deps\\meteormd-591af4f6383d7032.3cd6chswfvd2kjg0.rcgu.o" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\meteormd-5\CLionProjects\\meteormd\\target\\debug\\deps\\meteormd-591af4f6383d7032.4gbjbv4rjndh8kmi.rcgu.o" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\meteormd-591af4f6rojects\\meteormd\\target\\debug\\deps\\meteormd-591af4f6383d7032.58d0x3adzpbesz31.rcgu.o" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\meteormd-591af4f6383d703meteormd\\target\\debug\\deps\\meteormd-591af4f6383d7032.ec82jsv400ro0m9.rcgu.o" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\meteormd-591af4f6383d7032.f0gxfxz6rmd\\target\\debug\\deps\\meteormd-591af4f6383d7032.exe" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\meteormd-591af4f6383d7032.2u6lfs610h9tjrhk.rcgu.o" "-Wl,--_sys-663964cfdfe2b3e7\\out" "-L" "C:\\Users\\JoshHannaford\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib" "-Wl,-Bstatic" "C:\\Users\\JoshHan.rlib" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\libregex_syntax-bfd9a89c20465b5b.rlib" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\demeteormd\\target\\debug\\deps\\libonig-c3dbb2eae216450c.rlib" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\libonig_sys-06ceb316dbbe7e9a.rlib" "C:\\Users\\JoshHab" "C:\\Users\\JoshHannaford\\CLionProjects\\meteormd\\target\\debug\\deps\\liblazy_static-0cbab4b7d5e5f26a.rlib" "-Wl,--start-group" "C:\\Users\\JoshHannaford\\.rustup\\toolchains\\nightl86_64-pc-windows-gnu\\lib\\libpanic_unwind-70e401e192adfa0a.rlib" "C:\\Users\\JoshHannaford\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\l3420bfad8d.rlib" "C:\\Users\\JoshHannaford\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libbacktrace-2aef07d1aa10623b.rlib" "C:\\Users\\Jons\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_demangle-a134aae683b053a7.rlib" "C:\\Users\\JoshHannaford\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\libcfg_if-169d95fb9b7d35ac.rlib" "C:\\Users\\JoshHannaford\\.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\liblibc-85a70f4968e8.rustup\\toolchains\\nightly-x86_64-pc-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\librustc_std_workspace_core-d7d7fde4e7b36578.rlib" "C:\\Users\\JoshHannaford\\.rustup\\toolchac-windows-gnu\\lib\\rustlib\\x86_64-pc-windows-gnu\\lib\\libcompiler_builtins-a47ce7c70dc89625.rlib" "-Wl,-Bdynamic" "-ladvapi32" "-lws2_32" "-luserenv" "-Wl,-Bstatic" "-lgcc_eh" "-lpthrea86_64-pc-windows-gnu\\lib\\rsend.o"
  = note: C:\Users\JoshHannaford\CLionProjects\meteormd\target\debug\deps\libonig_sys-06ceb316dbbe7e9a.rlib(regexec.o): In function `onig_setup_builtin_monitors_by_ascii_encoded_name':
          C:\Users\JoshHannaford\.cargo\registry\src\github.com-1ecc6299db9ec823\onig_sys-69.2.0/oniguruma/src/regexec.c:5787: undefined reference to `__imp___acrt_iob_func'


error: aborting due to previous error

error: could not compile `meteormd`.
```

The most important part looking to be this.
```
= note: C:\Users\JoshHannaford\CLionProjects\meteormd\target\debug\deps\libonig_sys-06ceb316dbbe7e9a.rlib(regexec.o): In function `onig_setup_builtin_monitors_by_ascii_encoded_name':
          C:\Users\JoshHannaford\.cargo\registry\src\github.com-1ecc6299db9ec823\onig_sys-69.2.0/oniguruma/src/regexec.c:5787: undefined reference to `__imp___acrt_iob_func'
```

Am I doing something wrong?

Thanks for any help!!

**Comment by iwillspeak** [comment]
I think this is down to the toolchain that built the c code being different to the one that is building the Rust and performing the final link.

This is an incompatibility in the mingw/msys took gains that I vaguely remember fixing at one point. Have you looked at the onig CI build and how it sets the toolchains up?

**Comment by iwillspeak** [comment]
I have a feeling that this is the related part: https://github.com/rust-onig/rust-onig/blob/6bd5964978813dd7a7be9ce9604c0c6e76f4fe02/.appveyor.yml#L86

It might be worth applying the work around from the first issue in this project so other users don’t hit this same issue in the future. I’ll track addressing rust-lang/rust#47048 with this issue.

**Comment by Cy-Tek** [comment]
Thanks for reaching out! That is the issue indeed, I tried the workaround suggested in that issue and it resolved the problem for me. That's at least a good stopgap measure for now, so thank you for pointing me there.

Should I close this issue or leave it open for your tracking?

**Comment by iwillspeak** [comment]
Leave it open please. Now I know that there is a work around I’ll try and
add it to this crate.

**Comment by iwillspeak** [comment]
Ok. This shoudl be fixed upstream in rust-lang/rust#67429
